test_conformance/extensions/cl_khr_external_semaphore/test_external_semaphore.cpp

```diff
@@ -104,48 +114,60 @@ int test_external_semaphores_queries(cl_device_id deviceID, cl_context context,
     GET_PFN(deviceID, clReleaseSemaphoreKHR);
     GET_PFN(deviceID, clRetainSemaphoreKHR);
 
-    const std::vector<VulkanExternalMemoryHandleType>
-        vkExternalMemoryHandleTypeList =
-            getSupportedVulkanExternalMemoryHandleTypeList();
-    VulkanExternalSemaphoreHandleType vkExternalSemaphoreHandleType =
-        getSupportedVulkanExternalSemaphoreHandleTypeList()[0];
-    VulkanSemaphore vkVk2CLSemaphore(vkDevice, vkExternalSemaphoreHandleType);
+    std::vector<VulkanExternalSemaphoreHandleType>
+        vkExternalSemaphoreHandleTypeList =
+            getSupportedInteropExternalSemaphoreHandleTypes(deviceID, vkDevice);
 
-    clExternalSemaphore sema_ext(vkVk2CLSemaphore, context,
-                                 vkExternalSemaphoreHandleType, deviceID);
+    if (vkExternalSemaphoreHandleTypeList.empty())
+    {
+        test_fail("No external semaphore handle types found\n");
+    }
 
-    // Needed by the macro
-    cl_semaphore_khr sema = sema_ext.getCLSemaphore();
+    for (VulkanExternalSemaphoreHandleType vkExternalSemaphoreHandleType :
+         vkExternalSemaphoreHandleTypeList)
+    {
+        log_info_semaphore_type(vkExternalSemaphoreHandleType);
+        VulkanSemaphore vkVk2CLSemaphore(vkDevice,
+                                         vkExternalSemaphoreHandleType);
 
-    SEMAPHORE_PARAM_TEST(CL_SEMAPHORE_TYPE_KHR, cl_semaphore_type_khr,
-                         CL_SEMAPHORE_TYPE_BINARY_KHR);
+        clExternalSemaphore sema_ext(vkVk2CLSemaphore, context,
+                                     vkExternalSemaphoreHandleType, deviceID);
 
-    SEMAPHORE_PARAM_TEST(CL_DEVICE_HANDLE_LIST_KHR, cl_device_id, deviceID);
+        // Needed by the macro
+        cl_semaphore_khr sema = sema_ext.getCLSemaphore();
 
-    SEMAPHORE_PARAM_TEST(
-        CL_SEMAPHORE_EXPORT_HANDLE_TYPES_KHR, cl_uint,
-        getCLSemaphoreTypeFromVulkanType(vkExternalSemaphoreHandleType));
+        SEMAPHORE_PARAM_TEST(CL_SEMAPHORE_TYPE_KHR, cl_semaphore_type_khr,
+                             CL_SEMAPHORE_TYPE_BINARY_KHR);
 
-    // Confirm that querying CL_SEMAPHORE_CONTEXT_KHR returns the right context
-    SEMAPHORE_PARAM_TEST(CL_SEMAPHORE_CONTEXT_KHR, cl_context, context);
+        SEMAPHORE_PARAM_TEST(CL_DEVICE_HANDLE_LIST_KHR, cl_device_id, deviceID);
 
-    // Confirm that querying CL_SEMAPHORE_REFERENCE_COUNT_KHR returns the right
-    // value
-    SEMAPHORE_PARAM_TEST(CL_SEMAPHORE_REFERENCE_COUNT_KHR, cl_uint, 1);
+        SEMAPHORE_PARAM_TEST(
+            CL_SEMAPHORE_EXPORT_HANDLE_TYPES_KHR, cl_uint,
```

Comment:
A couple of things -
1. If we are creating semaphore by importing it from Vulkan, export handle types list can be NULL/absent. So, this place doesn't seem about right for me to check for exportable handle types query.
2. The return type for CL_SEMAPHORE_EXPORT_HANDLE_TYPES_KHR query is cl_external_semaphore_handle_type_khr[], not cl_uint.

Reply:
Agreed.  This isn't new code for my change.  Lets raise an issue to ensure we properly test this query, I would rather not add a new subtest in this change.